lib/pacemaker/pcmk_resource.c

```diff
@@ -41,7 +40,7 @@ best_op(const pe_resource_t *rsc, const pe_node_t *node,
 
     // Find node's resource history
     xpath = crm_strdup_printf(XPATH_OP_HISTORY, node->details->uname, rsc->id);
-    history = get_xpath_object(xpath, data_set->input, LOG_NEVER);
+    history = get_xpath_object(xpath, rsc->cluster->input, LOG_NEVER);
```

Comment:
not sure whether this has side effects besides via `getDocPtr()`. If so, then `rsc` is technically [in,out] and non-const.

Reply:
get_xpath_object() is safe other than getDocPtr() (complicated by libxml2 not using const where it could)